### DISCUSSION TITLE
Vim Copilot not working for Javascript/Typescript files

### DISCUSSION THREAD

**Discussion by jorgerojas26** [community/community#16593] [Copilot Conversations] [answered]
Suggestions are not showing up on Javascript and Typescript files only.

I've tried with Lua, HTML, and CSS and it's working properly.

Neovim 0.7.0
Vim Copilot 1.3
OS: Mac

I have read the log files and I do not find any error, everything is setup correctly.

:Copilot status
`Copilot : Enabled and online`

It seems to me that suggestions are just not being received by the client.

Is anyone facing this problem too?

Edit:
Just to be clear, copilot is working for filetypes other than Javascript/Typescript. The issue is only with those files.

**Comment by davecheney** [comment]
@jorgerojas26 it looks like you were using an older version of the copilot.vim client, but have since upgraded so it should be working now.

**Reply by jorgerojas26** [reply]
It is not working, i did update to vim copilot 1.3.0 (most recent version) and it is not working.

I did test downgrading to older versions but nothing changes.

**Reply by davecheney** [reply]
Is it possible that your employer, internet service provider, or government, may be intercepting your SSL traffic?

**Reply by jorgerojas26** [reply]
That's possible, but again, copilot is working for other filetypes like .lua, .html, .css. The issue is with Javascript/Typescript only, as far as i tested.

**Comment by jorgerojas26** [comment, marked as answer]
This stack overflow post has the solution.

https://stackoverflow.com/questions/72174839/solved-github-coptilot-does-not-work-in-neovim-when-editing-python

It was the node version. If you have node 18, downgrade to 17.9.0, that will fix it.

**Reply by tpope** [reply]
Can you tell me _how_ you installed Node? I just tested with 18.0.0 from Brew and it seems to work fine.

**Reply by jorgerojas26** [reply]
I used nvm.

**Reply by jorgerojas26** [reply]
I have multiple versions of node on my machine, that's why I use nvm.

I use the v14.15.0 as the default but for copilot I was using 18.1.0, pointing to it with the vim global variable:

 let g:copilot_node_command = '/Users/username/.nvm/versions/node/v18.1.0/bin/node'

Then, I installed node 17.9.0 from nvm, updated the global variable and immediately started to work.

I was using node 18.1.0 (not 18.0.0). So, maybe there is a problem with 18.1, or maybe the problem is with nvm.

I give you the info so maybe you can test better and get the exact problem.

**Reply by tpope** [reply]
Can indeed reproduce the issue on Node 18.1.0, regardless of installation method. I endorse the workaround of using an older Node version until this is resolved.

**Comment by sigmaSd** [comment]
Here is my personal workaround:
- Add this to AgentRequest in agent.vim
```vimscript
  "hardcode languageId to a working language
    if has_key(request.params, 'doc')
        let request.params.doc.languageId = "perl"
    endif
```
